src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -632,6 +630,29 @@ def run_acr_build(self, dockerfile, source, quiet=False, build_from_source=False
                 dockerfile,
                 quiet,
             )
+            return False
+
+        # Only enable Cloud Build on Stage and Canary while the changes are deployed to all regions.
+        is_cloud_build_enabled = self.env.location and any(self.env.location.lower() == region for region in ["northcentralusstage", "centraluseuap", "eastus2euap"])
```

Comment:
nit: should we extract out this location array into a separate variable that we can easily update as cloud build is enabled for additional locations? Preferably in some constants file, or some file that is related to the cloud build service

Reply:
So this is only a thing for this week/next week, as we need to deploy things to get some validation while deploying to more regions. As soon as we have the validation we need, this check will be removed.